components/User/UserProfile/PrivatePublicUserProfile/UserDecks.js

```diff
@@ -1,48 +1,116 @@
+import { fetchNextUserDecks } from '../../../../actions/user/userprofile/fetchNextUserDecks';
 
 class UserDecks extends React.Component {
     constructor(props){
         super(props);
-        this.sortBy = '2';
+        this.messages = this.getIntlMessages();
     }
     componentDidMount() {
         $(this.refs.sortDropdown).dropdown({onChange: this.dropdownSelect.bind(this)});
     }
 
-    componentDidUpdate() {}
+    componentDidUpdate() {
+        $(this.refs.sortDropdown).dropdown({onChange: this.dropdownSelect.bind(this)});
+    }
 
     dropdownSelect(value) {
-        this.sortBy = value;
-        this.forceUpdate();
+        this.context.executeAction(fetchUserDecks, {
+            params: {
+                username: this.props.user.uname, 
+                sort: value,
+                roles: 'owner',
+            }
+        });
+    }
+    loadMore(nextLink){       
+        this.context.executeAction(fetchNextUserDecks, {
+            nextLink: nextLink
+        });
+    }
+    getIntlMessages(){
+        return defineMessages({
+            sortLastUpdated: {
+                id: 'UserDecks.sort.lastUpdated', 
+                defaultMessage: 'Last updated'
+            }, 
+            sortCreationDate: {
+                id: 'UserDecks.sort.date', 
+                defaultMessage: 'Creation date'
+            },
+            sortTitle: {
+                id: 'UserDecks.sort.title', 
+                defaultMessage: 'Title'
+            }
+        });
+    }
+    getSelectedSort(sortBy){
+        switch(sortBy){
+            case 'timestamp': 
+                return this.context.intl.formatMessage(this.messages.sortCreationDate);
+            case 'title':
+                return this.context.intl.formatMessage(this.messages.sortTitle);
+            case 'lastUpdate': 
+            default: 
+                return this.context.intl.formatMessage(this.messages.sortLastUpdated);
+        }
     }
-
     render() {
+         // define load more results div
+        let loadMoreDiv = '';
+        let meta = this.props.decksMeta;
+        if(meta.links && meta.links.next){
+            let loadMoreContent = <button className="ui button" aria-label='Load more decks' onClick={this.loadMore.bind(this, meta.links.next)}>Load More</button>;
+            if(this.props.loadMoreLoading){
+                loadMoreContent = <div className="ui active text loader">Loading</div>;
+            }
+            if(this.props.loadMoreError){
+                loadMoreContent = 'An unexpected error occurred while fetching more decks';
+            }
+            loadMoreDiv = <div key="loadMoreDiv" className="ui basic segment center aligned">
+                {loadMoreContent}
+            </div>;
+        }
+        let sortBy = meta.sort;
+        let header = (this.props.loggedinuser === this.props.user.uname) 
+            ? <FormattedMessage id='UserDecks.header.myDecks' defaultMessage='My Decks' />
```

Comment:
Please exchange FormattedMessage as I often have errors with it.

Reply:
With this.context.intl.formatMessage of course

Reply:
Thanks for pointing it. I also remember having problems with it especially using it inside dropdown menus.